pkg/monitor/alerts.go

```diff
@@ -32,11 +36,24 @@ func whenWasAlertInState(ctx context.Context, prometheusClient prometheusv1.API,
 	if alertState != "pending" && alertState != "firing" {
 		return nil, fmt.Errorf("unrecognized alertState: %v", alertState)
 	}
+
+	// If the time range exceeds the Prometheus hardcoded 11,000 seconds, calculate an interval
+	// that Prometheus can handle.  This situation occurs when the testing time exceeds 3h3m20s
+	// (i.e., 11,000 seconds) which indicates something is taking a long time. Increasing the
+	// interval allows the alert tests to still run.
+	timeNow := time.Now()
+	queryInterval := 1
+	if timeNow.Sub(startTime).Seconds() > prometheusMaxResolution {
+		queryInterval = int(timeNow.Sub(startTime).Seconds())/prometheusMaxResolution + 1
+		fmt.Printf("Warning: Alert test range exceeds %d seconds ; using range interval of %d\n", prometheusMaxResolution, queryInterval)
+	}
+
 	timeRange := prometheusv1.Range{
 		Start: startTime,
-		End:   time.Now(),
-		Step:  1 * time.Second,
```

Comment:
@deads2k or @simonpasquier looking for opinions on this.

We will exceed 11,000 seconds when a test exceeds 3 hours (and this should not happen normally).  I use a calculation to ensure the Steps fit inside the 11,000 limit but we were wondering if it makes sense to just change the Step to `2 * time.Second` in general.  This would cover the cases for anything up to 6 hours (which is well beyond Prow's 4 hour job limit) and keep the Step consistent for tests that run for less than 3 hours as well as greater than 3 hours.

What do you think? thanks.

Reply:
Thanks for the comment @simonpasquier I updated my change.